src/RoleRegistry.sol

```diff
@@ -1,50 +1,89 @@
-    //------------------------------------  ADMIN ------------------------------------------
-    //--------------------------------------------------------------------------------------
+    /// @notice Checks if an account has any of the specified roles
+    /// @dev Reverts if the account doesn't have at least one of the roles
+    /// @param account The address to check roles for
+    /// @param encodedRoles ABI encoded roles (abi.encode(ROLE_1, ROLE_2, ...))
+    function checkRoles(address account, bytes memory encodedRoles) public view {
+        if (!_hasAnyRoles(account, encodedRoles)) __revertEnumerableRolesUnauthorized();
+    }
 
-    /// @notice sets the target role to be managed by another role
-    /// @dev only the overall admin has the ability to update role admins
-    /// @param _targetRole is the role you are changing the admin of
-    /// @param _adminRole is the role that will be the new admin of the _targetRole
-    function setRoleAdmin(bytes32 _targetRole, bytes32 _adminRole) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        _setRoleAdmin(_targetRole, _adminRole);
+    /// @notice Checks if an account has a specific role
+    /// @param role The role to check (as bytes32)
+    /// @param account The address to check the role for
+    /// @return bool True if the account has the role, false otherwise
+    function hasRole(bytes32 role, address account) public view returns (bool) {
+        return hasRole(account, uint256(role));
     }
 
-    function _authorizeUpgrade(address newImplementation) internal override onlyRole(DEFAULT_ADMIN_ROLE) {}
+    /// @notice Grants a role to an account
+    /// @dev Only callable by the contract owner (handled in setRole function)
+    /// @param role The role to grant (as bytes32)
+    /// @param account The address to grant the role to
+    function grantRole(bytes32 role, address account) public {
+        setRole(account, uint256(role), true);  
+    } 
+
+    /// @notice Revokes a role from an account
+    /// @dev Only callable by the contract owner (handled in setRole function)
+    /// @param role The role to revoke (as bytes32)
+    /// @param account The address to revoke the role from
+    function revokeRole(bytes32 role, address account) public {
+        setRole(account, uint256(role), false);  
+    }
+
+    /// @notice Gets all addresses that have a specific role
+    /// @dev Wrapper around EnumerableRoles roleHolders function converting bytes32 to uint256
+    /// @param role The role to query (as bytes32)
+    /// @return address[] Array of addresses that have the specified role
+    function roleHolders(bytes32 role) public view returns (address[] memory) {
+        return roleHolders(uint256(role));
+    }
+
+    function onlyProtocolUpgrader(address account) public view {
+        if (owner() != account) revert OnlyProtocolUpgrader();
+    }
+
+    function __revertEnumerableRolesUnauthorized() private pure {
```

Comment:
is there a reason we implemented this ourselves? I imagine it is part of the base implementation